### ISSUE TITLE
Create a left and right topbar section (maybe a middle one as well?)

### ISSUE THREAD

**Issue by choldgraf** [pradyunsg/sphinx-basic-ng#6]
It is common to have topbar content that is both justified to the left *and* justified to the right. Other themes also include a "center topbar" section that aligns with the page content.

How could we accommodate these different workflows?

For example:

pydata sphinx theme:

![image](https://user-images.githubusercontent.com/1839645/111083635-b97aea80-84cb-11eb-9ded-fe503c759431.png)

gitbook theme:

![image](https://user-images.githubusercontent.com/1839645/111083659-db746d00-84cb-11eb-9c16-43fde3e5a555.png)

quantecon theme:

![image](https://user-images.githubusercontent.com/1839645/111083673-eaf3b600-84cb-11eb-9deb-9842e82a268e.png)

**Comment by pradyunsg** [comment]
Yup yup. My thought process is that this would be what the default wrapper would provide. I just didn't have enough time to write it all down. 🙃

**Comment by pradyunsg** [comment]
Bringing this over from the above issue:

> I think the header should be split into left_header and right_header template sections.

Actually, I had a bit of a different idea for this:

- If there's 2 tags (i.e. elements) in the header component, they would be left-right aligned.
- If there's 3 elements in the header component, they would be left-center-right aligned.

This is much easier to maintain upstream thanks to fewer points to interact with downstream themes and it is easier to adapt for RTL languages as well.

**Comment by choldgraf** [comment]
interesting - that seems reasonable from an end-user standpoint, though I must admit that it seems complicated to implement from my perspective. If you think it's more straightforward though, then go for it :-)

**Comment by pradyunsg** [comment]
Flexbox FTW!

https://css-tricks.com/snippets/css/a-guide-to-flexbox/#justify-content

Just need to have `justify-content: space-between`. :)

**Comment by choldgraf** [comment]
wow that's cool - so then from a theme developer's standpoint, you'd need to document that "any `components` put in the topbar should use the `justify-content` CSS rule"? That's not something that could automatically be done for people right?

**Comment by pradyunsg** [comment]
The `sections/topbar.html` file would be:


```
<div>This is aligning left</div>
<div>This is aligning right</div>
```

Or

```
<div>This is aligning left</div>
<div>This is aligning center</div>
<div>This is aligning right</div>
```

And they'll get aligned as the names suggest. :)

**Comment by choldgraf** [comment]
makes sense - so basically a sub-theme would:

1. Override `sections/topbar.html` in their template.
2. If they added two divs like:

   ```
   <div>{{ include some templates }}</div>
   <div>{{ include some templates }}</div>
   ```

   Then they'd be in two divs that take up the left and right half of the screen.

and if they added in a third div, then the middle div would be middle-centered.

or maybe more generally the divs would have the same width and spread evenly?

**Comment by pradyunsg** [comment]
> and if they added in a third div, then the middle div would be middle-centered.

Yes, if the left and right are the same width. Otherwise the middle item will be slightly off.

> or maybe more generally the divs would have the same width and spread evenly?

Not the width but yea, spread evenly.
